src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java

```diff
@@ -299,6 +299,22 @@ void testGeometryMapping() {
         assertVirtualTableContents(table, table("VARCHAR").row("POINT (2 3)").matches());
     }
 
+    @Test
+    void testDecimalLiteral() {
+        final Table table = createSingleColumnTable("BOOLEAN").insert(true);
+        this.virtualSchema = createVirtualSchema(this.sourceSchema);
+        assertVsQuery("SELECT 10.2 FROM " + getVirtualTableName(this.virtualSchema, table),
+                table("DECIMAL").row(BigDecimal.valueOf(10.2)).matches());
+    }
+
+    @Test
+    void testDoubleLiteral() {
+        final Table table = createSingleColumnTable("BOOLEAN").insert(true);
+        this.virtualSchema = createVirtualSchema(this.sourceSchema);
+        assertVsQuery("SELECT CAST(10.2 as DOUBLE) FROM " + getVirtualTableName(this.virtualSchema, table),
+                table("DOUBLE PRECISION").row(10.2).matches());
```

Comment:
So the idea is to cast to double when corresponding Exasol column is `double precision`?

Reply:
yes. It's a workaround for Exasol to being not standard conform in SQL parsing.